test/jdk/com/sun/net/httpserver/Test12.java

```diff
@@ -49,27 +38,32 @@
 import static jdk.test.lib.Asserts.assertFileContentsEqual;
 import static jdk.test.lib.Utils.createTempFileOfSize;
 
-/* basic http/s connectivity test
- * Tests:
- *      - same as Test1, but in parallel
+/*
+ * @test
+ * @bug 6270015 8359477
+ * @summary Light weight HTTP server - basic http/s connectivity test, same as Test1,
+ *          but in parallel
+ * @library /test/lib
+ * @build jdk.test.lib.Asserts
+ *        jdk.test.lib.Utils
+ *        jdk.test.lib.net.SimpleSSLContext
+ *        jdk.test.lib.net.URIBuilder
+ * @run main/othervm Test12
+ * @run main/othervm -Djava.net.preferIPv6Addresses=true Test12
  */
-
 public class Test12 extends Test {
 
     private static final String TEMP_FILE_PREFIX =
             HttpServer.class.getPackageName() + '-' + Test12.class.getSimpleName() + '-';
 
     static SSLContext ctx;
 
-    static boolean fail = false;
-
     public static void main (String[] args) throws Exception {
         HttpServer s1 = null;
         HttpsServer s2 = null;
-        ExecutorService executor=null;
         Path smallFilePath = createTempFileOfSize(TEMP_FILE_PREFIX, null, 23);
         Path largeFilePath = createTempFileOfSize(TEMP_FILE_PREFIX, null, 2730088);
-        try {
+        try (final ExecutorService executor = Executors.newCachedThreadPool()) {
```

Comment:
I would not use try-with-resources here: I don't think we should close the executor before closing the servers.

Reply:
I've updated the PR to close the Executor after the servers have been closed. I'm guessing the suggestion was mostly as a precaution than what this test currently does?